superset-frontend/src/explore/components/controls/SelectControl.jsx

```diff
@@ -94,8 +94,8 @@ export default class SelectControl extends React.PureComponent {
 
   UNSAFE_componentWillReceiveProps(nextProps) {
     if (
-      nextProps.choices !== this.props.choices ||
-      nextProps.options !== this.props.options
+      !isEqualArray(nextProps.choices, this.props.choices) ||
+      !isEqualArray(nextProps.options, this.props.options)
```

Comment:
Had to run this deeper check because mapStateToProps may create columns list with a new array but same content. We may need to dig deeper to those excessive rerendering and check whether they are actually necessary.